source/core/Variants.cpp

```diff
@@ -487,6 +487,11 @@ bool VariantsAreEqual(VariantTypeRef variantX, VariantTypeRef variantY)
 {
     DualTypeVisitor visitor;
     DualTypeEqual dualTypeEqual;
+    if (!variantX && !variantY) {
```

Comment:
Ideally, this check should happen inside the Visitor's Visit method when we handle Variant compatibility. However, since the TypeRefs are `nullptr` in case of empty variant constants, calling Begin(read) on it is not possible. 

Additionally, we will have to revisit this once nested variants are supported anyway.